<!DOCTYPE html>
<html lang="en"><head>
<script src="Intro_to_bioinformatics2023_files/libs/clipboard/clipboard.min.js"></script>
<script src="Intro_to_bioinformatics2023_files/libs/quarto-html/tabby.min.js"></script>
<script src="Intro_to_bioinformatics2023_files/libs/quarto-html/popper.min.js"></script>
<script src="Intro_to_bioinformatics2023_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="Intro_to_bioinformatics2023_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Intro_to_bioinformatics2023_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="Intro_to_bioinformatics2023_files/libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.2.269">

  <meta name="author" content="and some R/bioinformatics">
  <title>Copy Number Variants and Cancer</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="Intro_to_bioinformatics2023_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="Intro_to_bioinformatics2023_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
      }
    pre.numberSource { margin-left: 3em;  padding-left: 4px; }
    div.sourceCode
      { color: #f8f8f2;  }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #f8f8f2; } /* Normal */
    code span.al { color: #f07178; background-color: #2a0f15; font-weight: bold; } /* Alert */
    code span.an { color: #d4d0ab; } /* Annotation */
    code span.at { color: #00e0e0; } /* Attribute */
    code span.bn { color: #d4d0ab; } /* BaseN */
    code span.bu { color: #abe338; } /* BuiltIn */
    code span.cf { color: #ffa07a; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #abe338; } /* Char */
    code span.cn { color: #ffd700; } /* Constant */
    code span.co { color: #f8f8f2; font-style: italic; } /* Comment */
    code span.cv { color: #ffd700; } /* CommentVar */
    code span.do { color: #f8f8f2; } /* Documentation */
    code span.dt { color: #ffa07a; } /* DataType */
    code span.dv { color: #d4d0ab; } /* DecVal */
    code span.er { color: #f07178; text-decoration: underline; } /* Error */
    code span.ex { color: #00e0e0; font-weight: bold; } /* Extension */
    code span.fl { color: #d4d0ab; } /* Float */
    code span.fu { color: #ffa07a; } /* Function */
    code span.im { color: #abe338; } /* Import */
    code span.in { color: #d4d0ab; } /* Information */
    code span.kw { color: #ffa07a; font-weight: bold; } /* Keyword */
    code span.op { color: #ffa07a; } /* Operator */
    code span.ot { color: #00e0e0; } /* Other */
    code span.pp { color: #dcc6e0; } /* Preprocessor */
    code span.re { color: #00e0e0; background-color: #f8f8f2; } /* RegionMarker */
    code span.sc { color: #abe338; } /* SpecialChar */
    code span.ss { color: #abe338; } /* SpecialString */
    code span.st { color: #abe338; } /* String */
    code span.va { color: #00e0e0; } /* Variable */
    code span.vs { color: #abe338; } /* VerbatimString */
    code span.wa { color: #dcc6e0; } /* Warning */
  </style>
  <link rel="stylesheet" href="Intro_to_bioinformatics2023_files/libs/revealjs/dist/theme/quarto.css" id="theme">
  <link href="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  
  <script src="Intro_to_bioinformatics2023_files/libs/kePrint-0.0.1/kePrint.js"></script>
  <link href="Intro_to_bioinformatics2023_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">
  <link rel="stylesheet" media="screen" href="https://fonts.googleapis.com/css2?family=Amatic+SC&amp;display=swap" type="text/css">
</head>
<body class="quarto-dark">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Copy Number Variants and Cancer</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
and some R/bioinformatics 
</div>
</div>
</div>

</section>
<section id="lecture-plan" class="slide level2">
<h2>Lecture Plan</h2>
<ul>
<li class="fragment">Background
<ul>
<li class="fragment">Genomic variation</li>
<li class="fragment">Copy number variants</li>
<li class="fragment">Genetic risk in Cancer</li>
<li class="fragment">Computational biology: Pro’s <fade>(and con’s)</fade></li>
</ul></li>
<li class="fragment">Case example: Endometrial cancer and copy number variants</li>
</ul>
<div class="cell" data-hash="Intro_to_bioinformatics2023_cache/revealjs/load_data_cfd1e61f3a1afe2c760505a6f9dcfc31">

</div>
<!-- . . . -->
<!-- [Cassie Stylianou]{class='csText'}  -->
<!-- [George Wiggins]{class='gwText'}  -->
<!-- ![Cassie Stylianou](images/cass.jpg){.cass} ![George Wiggins](images/cropped.jpg){.george} -->
</section>
<section id="genomic-variation" class="slide level2">
<h2>Genomic variation</h2>

<img data-src="images/typeofvariants.png" class="r-stretch"></section>
<section id="copy-number-variants" class="slide level2">
<h2>Copy number variants</h2>
<div>
<ul>
<li><p>Copy number variants (CNV) are a subset of structural variants</p>
<ul>
<li>CNVs are the number of copies of a DNA loci (i.e.&nbsp;genes) that vary between individuals.</li>
</ul></li>
<li><p>10% of the human genome contributes to CNVs<sup>1</sup>, 1% of human genome contributes to SNPs.</p></li>
<li><p>Rare CNV (&lt;1% MAF) are 800x more likely to be deleterious compared to rare SNV<sup>2</sup>.</p></li>
</ul>
<div class="reference">
<p><sup>1</sup> Zarrei et al.&nbsp;(2015) Nat Rev Genet. 16, 172-183.</p>
<p><sup>2</sup> HJ Abel et al.&nbsp;(2020). Nature. 583,83-89.</p>
<!-- PJ Hastings et al. (2009) Nat Rev Genet. 10,551-564.  -->
</div>
</div>
</section>
<section id="cnv-detection" class="slide level2">
<h2>CNV detection</h2>
<div style="margin-left:-100px; width:57.5%">
<ol type="1">
<li>Fluorescence in situ hybridization (FISH)</li>
<li>Array Comparative genomic hybridization (CGH)</li>
<li>qPCR</li>
<li>MLPA</li>
<li>SNP array</li>
<li>Next-generation sequencing</li>
</ol>
</div>
<p><img data-src="images/CNV_detection.png" class="absolute" style="top: 50px; right: -100px; width: 55%; "></p>
</section>
<section id="section" class="slide level2">
<h2></h2>
<div class="quote">
<p>Cancer is the most common human genetic disease. The transition from a normal cell to a malignant cancer is driven by changes to a cell’s DNA.</p>
</div>
<p><img data-src="images/Tumor_progression.png" class="absolute" style="bottom: 0px; width: 1000px; "></p>
<!-- Individuals with cancer effectively have two (or more) genomes. **Germline** and **somatic** both are useful clinically -->
</section>
<section id="genetic-risk-of-cancer" class="slide level2">
<h2>Genetic Risk of Cancer</h2>
<p>DNA variants alter the risk of cancer.</p>
<p><img data-src="images/BC_risk_B1B2.png" class="absolute" style="top: 200px; left: 0px; width: 500px; "> <img data-src="images/B1_B2_risk.png" class="absolute" style="top: 200px; right: 20px; width: 450px; "></p>
<div class="reference">
<p>Kuchenbaecker et al.&nbsp;(2017) JAMA. 317(23), 2402-2416.</p>
</div>
</section>
<section id="genetic-risk-of-cancer-1" class="slide level2 smaller">
<h2>Genetic Risk of Cancer</h2>
<p>There are 313 common genetic variants associated with a low increase in breast cancer risk. There is up to a 4-fold increase in breast cancer for woman that harbour the greatest number of these risk allele<sup>1</sup>.</p>
<p><img data-src="images/BC_risk_common.png" class="absolute" style="left: 0px; width: 500px; "> <img data-src="images/polygenic.png" class="absolute" style="right: 150px; width: 320px; "></p>
<div class="reference">
<p><sup>1</sup> N. Zeinomar &amp; W.K. Chung (2020) Ann Intern Med. 174(3), 408-412.</p>
</div>
</section>
<section id="why-computational-biology" class="slide level2">
<h2>Why Computational Biology?</h2>
<div class="smaller-font">
<ul>
<li class="fragment">Lots of data (dataset sizes and databases)</li>
<li class="fragment">Visualisation - makes a big problem seem small</li>
</ul>
<div class="fragment">
<p><span class="description">Example: Gene expression data from <strong>25,000 genes</strong> for two cell lines with two treatments</span></p>
<p><img data-src="plots/pca_unlabelled.png" class="pca-left"></p>
</div>
<div class="fragment">
<p><img data-src="plots/pca_labelled.png" class="pca-right"></p>
</div>
</div>
</section>
<section id="why-computational-biology-1" class="slide level2">
<h2>Why Computational Biology?</h2>
<div class="smaller-font">
<div>
<ul>
<li>Lots of data (dataset sizes and databases)</li>
<li>Visualisation - makes a big problem seem small</li>
<li>Efficiency</li>
<li>New tools and methods - simulations, statistics</li>
</ul>
</div>
<div class="fragment">
<p>Barriers:</p>
</div>
<ul>
<li class="fragment">Can seem ‘foreign’ and difficult to learn</li>
<li class="fragment">Requires a solid statistical foundation (can be learnt on the fly)</li>
<li class="fragment">Can over-engineer solution</li>
</ul>
</div>
<!-- # Biology is Complex -->
<!-- ##  {background-image="images/bases.png" background-color="white" background-size="cover"} -->
<!-- ##  {background-image="images/bases_annotated.png" background-color="white" background-size="cover"} -->
</section>
<section id="omics-data-is-complex-and-large" class="slide level2 smaller">
<h2>Omics’ data is complex and LARGE</h2>
<table>
<caption>Types of omics’ data</caption>
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th></th>
<th style="text-align: center;">File Type</th>
<th style="text-align: right;">Size</th>
<th style="text-align: center;">No.&nbsp;Features</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Next Gen Sequencing</td>
<td style="text-align: center;">base calls (.fastq)</td>
<td style="text-align: right;">10-100 Gb</td>
<td style="text-align: center;">Up to everything</td>
</tr>
<tr class="even">
<td>Microarray</td>
<td style="text-align: center;">Intensities</td>
<td style="text-align: right;">100 Mb - 10 Gb</td>
<td style="text-align: center;">10,000s - millions</td>
</tr>
</tbody>
</table>
<p><img data-src="images/fastq1.png" class="absolute" style="left: 125px; bottom: 50px; width: 800px; "></p>
</section>
<section>
<section id="introduction-to-r" class="title-slide slide level1 center">
<h1>Introduction to R</h1>

</section>
<section id="r-and-rstudio" class="slide level2">
<h2>R and Rstudio <img src="images/R_logo.png" class="r" height="100px" style="vertical-align:middle;"> <img src="images/Rstudio.png" class="rstudio" height="100px" style="vertical-align:middle;"></h2>
<p>R is a programming language commonly used in the biological sciences.</p>
<div class="fragment">
<p>It is not the only suitable language (Python, C++, Bash)</p>
</div>
<div class="fragment">
<p><br>RStudio is an integrated development environment (IDE) for R. Basically, makes R more usable.</p>
</div>
</section>
<section id="r-terminology" class="slide level2">
<h2>R terminology</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1" data-code-line-numbers="1-4|6-7|9-10|12-13|15-19|21-24|26-29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="do">## variables</span></span>
<span id="cb1-2"><a href="#cb1-2"></a>x <span class="ot">&lt;-</span> <span class="dv">1</span> </span>
<span id="cb1-3"><a href="#cb1-3"></a>y <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)</span>
<span id="cb1-4"><a href="#cb1-4"></a>z <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"a"</span>,<span class="st">"b"</span>,<span class="st">"c"</span>)</span>
<span id="cb1-5"><a href="#cb1-5"></a></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="do">## matrices</span></span>
<span id="cb1-7"><a href="#cb1-7"></a>m <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="sc">-</span><span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">9</span>),<span class="at">nrow=</span><span class="dv">3</span>)</span>
<span id="cb1-8"><a href="#cb1-8"></a></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="do">## data.frames</span></span>
<span id="cb1-10"><a href="#cb1-10"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">col1=</span>z, <span class="at">col2=</span>y, <span class="at">col3=</span><span class="fu">c</span>(<span class="st">"a"</span>,<span class="dv">1</span>,<span class="st">"c"</span>))</span>
<span id="cb1-11"><a href="#cb1-11"></a></span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="do">## lists</span></span>
<span id="cb1-13"><a href="#cb1-13"></a>list.df <span class="ot">&lt;-</span> <span class="fu">list</span>(df,mtcars, iris)</span>
<span id="cb1-14"><a href="#cb1-14"></a></span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="do">## functions</span></span>
<span id="cb1-16"><a href="#cb1-16"></a><span class="fu">mean</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">23</span>,<span class="dv">43</span>))</span>
<span id="cb1-17"><a href="#cb1-17"></a><span class="fu">sd</span>(y)</span>
<span id="cb1-18"><a href="#cb1-18"></a>y2 <span class="ot">&lt;-</span> y<span class="sc">*</span><span class="fu">sample</span>(<span class="dv">5</span><span class="sc">:</span><span class="dv">50</span>,<span class="dv">1</span>) <span class="do">## multiple y by a random number between 5-50</span></span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="fu">t.test</span>(y,y2)</span>
<span id="cb1-20"><a href="#cb1-20"></a></span>
<span id="cb1-21"><a href="#cb1-21"></a><span class="do">## libraries (suites of functions)</span></span>
<span id="cb1-22"><a href="#cb1-22"></a><span class="fu">library</span>(tidyverse) <span class="do">## used for data wrangling</span></span>
<span id="cb1-23"><a href="#cb1-23"></a><span class="fu">library</span>(ggplot2) <span class="do">## nicer plotting</span></span>
<span id="cb1-24"><a href="#cb1-24"></a><span class="fu">library</span>(openxlsx) <span class="do">## read data store in as excel files</span></span>
<span id="cb1-25"><a href="#cb1-25"></a></span>
<span id="cb1-26"><a href="#cb1-26"></a><span class="do">## installing libraries(packages)</span></span>
<span id="cb1-27"><a href="#cb1-27"></a><span class="fu">install.packages</span>(<span class="st">'devtools'</span>) <span class="do">## if on CRAN</span></span>
<span id="cb1-28"><a href="#cb1-28"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"hpar"</span>) <span class="do">## if on bioconductor</span></span>
<span id="cb1-29"><a href="#cb1-29"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"yanlinlin82/ggvenn"</span>) <span class="do">## if on github</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section>
<section>
<section id="genetic-risk-of-endometrial-cancer-cassies-phd" class="title-slide slide level1 center">
<h1>Genetic Risk of Endometrial Cancer<br><span class="description">Cassie’s PhD</span></h1>

</section>
<section id="genetic-risk-small-variants" class="slide level2">
<h2>Genetic Risk: <span style="color:#068de9">Small variants</span></h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="plots/EC_risk.png" height="500"></p>
<p></p><figcaption>Dörk, T. et al.&nbsp;Cancers (2020), 12(9).</figcaption><p></p>
</figure>
</div>
</section>
<section id="genetic-risk-copy-number-variants" class="slide level2 smaller">
<h2>Genetic Risk: <span style="color:#068de9">Copy number variants</span></h2>
<p>Copy number variants are regions of genome where the number of copies differ between individuals.</p>
<p>How do these variant effect cancer risk?</p>
<div class="fragment fade-out">
<p><img data-src="images/CNV2.png" class="absolute" style="bottom: 100px; right: 10px; width: 300px; "></p>
</div>
<div class="fragment">
<div class="subheading">
<p>Cassie’s dataset</p>
</div>
<div class="columns">
<div class="column" style="width:60%;">
<ul>
<li>Woman genotyped with (case) and with out EC</li>
<li>CNVs called from array data (gene-centric) and associations study (GWAS) performed</li>
<li>3 GWAS types: Deletions, duplication and Loss of function</li>
</ul>
</div><div class="column" style="width:40%;">
<table>
<thead>
<tr class="header">
<th style="text-align: center;">-</th>
<th style="text-align: right;">Number of samples</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">Case</td>
<td style="text-align: right;">4,155</td>
</tr>
<tr class="even">
<td style="text-align: center;">Control</td>
<td style="text-align: right;">17,818</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="snp-array-and-cnvs" class="slide level2 smaller" data-auto-animate="true">
<h2 data-id="quarto-animate-title">SNP array and CNVs</h2>
<div style="right:-100px;position:absolute;width:45%;font-size: 0.9em;">
<p><span class="math inline">\(BAF = A/B\)</span></p>
<p><span class="math inline">\(Log R Ratio = log2(R~observed~/R~expected~)\)</span></p>
</div>
<div style="right:-100px;position:absolute;width:45%;top:190px">
<p>Deletion (example): Only one copy of DNA, therefore the observed intensity (R) is less than expected and all the signal will either be the A (BAF=0) or B (BAF=1) allele.</p>
</div>
<p><img data-src="plots/CNVvalling.png" class="absolute" style="top: 80px; left: -100px; width: 750px; "></p>
</section>
<section id="genome-wide-association-study" class="slide level2 smaller" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Genome-wide association study</h2>
<p><img data-src="plots/CNVvalling.png" class="absolute" style="top: 90px; left: 10px; width: 350px; "> <img data-src="images/flowchart.png" class="absolute" style="top: 80px; left: 30px; width: 1000px; "></p>
<!-- ## Data/results interrogation -->
<!-- ::: columns -->
<!-- ::: {.column width="65%"} -->
<!-- Do we believe the results? -->
<!-- Can the results be explained away by limitations/biases? -->
<!-- ::: -->
<!-- ::: {.column style="font-size:1.1rem; width:35%"} -->
<!-- [GWAS results]{style="font-size:2.5rem;"} -->
<!-- | CNV type | No. genes tested | -->
<!-- |----------|-----------------:| -->
<!-- | Del      |            5,608 | -->
<!-- | Dup      |            7,887 | -->
<!-- | LoF      |            8,071 | -->
<!-- ::: -->
<!-- ::: -->
<!-- . . . -->
<!-- [Let's ask some *very basic* questions:]{style="color:#068de9"} -->
<!-- -   How many associations were significant? -->
<!-- -   How many genes were tested across by all three GWAS? -->
<!-- -   How many significant association overlap (Venn Diagram)? -->
<!-- ## Excel vs R {.smaller} -->
<!-- How many associations were significant? -->
<!-- . . . -->
<!-- Excel (easy): -->
<!-- -   filter by p-value ➙ count rows -->
<!--     -   repeat per GWAS -->
<!-- ::: {.fragment .fade-in} -->
<!-- R (easy/over-engineered): -->
<!-- ```{r eval=F, echo=T} -->
<!-- #| code-line-numbers: "1-7|9" -->
<!-- library(openxlsx) -->
<!-- library(tidyverse) -->
<!-- ## Read in data -->
<!-- sheetNames <- getSheetNames("data/GWAS_3probe.xlsx") -->
<!-- df <- lapply(sheetNames, function(i) read.xlsx("data/GWAS_3probe.xlsx", sheet = i)) -->
<!-- names(df) <- sheetNames -->

<!-- df<-data.table::rbindlist(df, fill=T, idcol = "GWAS") -->
<!-- df %>% group_by(GWAS) %>% summarise(n=length(P_value <0.05)) -->
<!-- ``` -->
<!-- ::: -->
<!-- ## Excel vs R {.smaller} -->
<!-- How many genes were tested across by all three GWAS? -->
<!-- . . . -->
<!-- Excel (moderate/cumbersome): -->
<!-- 1.  Create unique list of all genes -->
<!-- 2.  `=countif(AllGenesinGWAS1, gene)` -->
<!--     -   repeat per GWAS -->
<!-- 3.  Sum counts ➙ filter and count rows equal to three -->
<!-- ::: {.fragment .fade-in} -->
<!-- R (easy): -->
<!-- ```{r eval=F, echo=T} -->
<!-- #| code-line-numbers: "9" -->
<!-- library(openxlsx) -->
<!-- library(tidyverse) -->
<!-- ## Read in data -->
<!-- sheetNames <- getSheetNames("data/GWAS_3probe.xlsx") -->
<!-- df <- lapply(sheetNames, function(i) read.xlsx("data/GWAS_3probe.xlsx", sheet = i)) -->
<!-- names(df) <- sheetNames -->

<!-- df<-data.table::rbindlist(df, fill=T, idcol = "GWAS") -->
<!-- sum(table(df$Gene.ID) ==3) # only line to change -->
<!-- ``` -->
<!-- ::: -->
<!-- ## Excel vs R {.smaller} -->
<!-- How many significant association overlap (Venn Diagram)? -->
<!-- . . . -->
<!-- Excel (Difficult): -->
<!-- -   Nested countifs with multiple criteria? -->
<!-- -   Is a very manual process -->
<!-- ## Excel vs R {.smaller} -->
<!-- How many significant association overlap (Venn Diagram)? -->
<!-- R (easy): -->
<!-- ```{r tri-venn, eval=T, echo=T} -->
<!-- #| output-location: column -->
<!-- #| code-line-numbers: "3,11-18" -->
<!-- #| cache: true -->
<!-- #| fig-height: 7.5 -->
<!-- library(openxlsx) -->
<!-- library(tidyverse) -->
<!-- library(ggvenn) ## New package -->
<!-- ## Read in data -->
<!-- sheetNames <- getSheetNames("data/GWAS_3probe.xlsx") -->
<!-- df <- lapply(sheetNames, function(i) read.xlsx("data/GWAS_3probe.xlsx", sheet = i)) -->
<!-- names(df) <- sheetNames -->

<!-- df<-data.table::rbindlist(df, fill=T, idcol = "GWAS") -->
<!-- x<-list( -->
<!--   DEL = df$Gene.ID[df$P_value<0.05 & df$GWAS == "DEL"],  -->
<!--   DUP = df$Gene.ID[df$P_value<0.05 & df$GWAS == "DUP"], -->
<!--   LOF = df$Gene.ID[df$P_value<0.05 & df$GWAS == "LOF"]  -->
<!-- ) -->
<!-- ggvenn(x,fill_color = c("#0073C2FF", "#EFC000FF",  "#CD534CFF"), -->
<!--   stroke_size = 1, set_name_size = 8, text_size = 8, show_percentage = F) -->
<!-- ``` -->
</section></section>
<section>
<section id="hypothetical-whole-gene-duplications-and-endometiral-cancer-risk" class="title-slide slide level1 center">
<h1><span style="color: #d56262;">HYPOTHETICAL:</span><br> <span class="question-heading">Whole gene duplications and Endometiral Cancer risk?</span></h1>

</section>
<section id="approach" class="slide level2">
<h2>Approach</h2>
<p>Cassie’s loss of function data used CNV predicted that:</p>
<ol type="1">
<li class="fragment">Delete a gene (part of or entire gene)</li>
<li class="fragment">Partially duplicated a gene (<strong>not</strong> whole gene)</li>
</ol>
<div class="fragment fade-in-then-out">
<p><img data-src="images/LoF_selection.png" class="absolute" style="left: 225px; bottom: 180px; width: 520px; "></p>
</div>
<div class="fragment">
<p>Duplication are frequently in tandem, therefore a partial gene duplication would disrupt the coding sequence.</p>
</div>
<div class="fragment">
<p><br></p>
<p><em>Only the ‘duplication’ GWAS tested whole gene duplications.</em></p>
</div>
</section>
<section id="whole-gene-duplications" class="slide level2 smaller nonincremental">
<h2>Whole gene duplications</h2>
<ol type="1">
<li class="fragment">Find significant associations (genes) unique to the duplication GWAS <span style="color:#068de9">(i.e.&nbsp;whole gene duplications)</span></li>
</ol>
<div class="fragment">
<div class="columns">
<div class="column" style="width:65%;">
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="do">## Read in data</span></span>
<span id="cb2-4"><a href="#cb2-4"></a>sheetNames <span class="ot">&lt;-</span> <span class="fu">getSheetNames</span>(<span class="st">"data/GWAS_3probe.xlsx"</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a>df <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sheetNames, <span class="cf">function</span>(i) <span class="fu">read.xlsx</span>(<span class="st">"data/GWAS_3probe.xlsx"</span>, <span class="at">sheet =</span> i))</span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="fu">names</span>(df) <span class="ot">&lt;-</span> sheetNames</span>
<span id="cb2-7"><a href="#cb2-7"></a>df<span class="ot">&lt;-</span>data.table<span class="sc">::</span><span class="fu">rbindlist</span>(df, <span class="at">fill=</span>T, <span class="at">idcol =</span> <span class="st">"GWAS"</span>)</span>
<span id="cb2-8"><a href="#cb2-8"></a></span>
<span id="cb2-9"><a href="#cb2-9"></a>df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(GWAS <span class="sc">!=</span> <span class="st">"DEL"</span>, P_value <span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-10"><a href="#cb2-10"></a>  <span class="fu">group_by</span>(Gene_symbol) <span class="sc">%&gt;%</span> </span>
<span id="cb2-11"><a href="#cb2-11"></a>  <span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>(), GWAS) <span class="sc">%&gt;%</span></span>
<span id="cb2-12"><a href="#cb2-12"></a>  <span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">1</span>, GWAS <span class="sc">==</span> <span class="st">"DUP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">

<table style="width:100%;margin-top:20px; font-size: 20px; margin-left: auto; margin-right: auto;" class="table">
 <thead>
  <tr>
   <th style="text-align:left;"> Gene Symbol </th>
   <th style="text-align:left;"> p value </th>
   <th style="text-align:right;"> Cases </th>
   <th style="text-align:right;"> Controls </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> FRMD1 </td>
   <td style="text-align:left;"> 1.78e-05 </td>
   <td style="text-align:right;"> 119 </td>
   <td style="text-align:right;"> 327 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> GPRIN2 </td>
   <td style="text-align:left;"> 2.28e-04 </td>
   <td style="text-align:right;"> 34 </td>
   <td style="text-align:right;"> 68 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> CRHR1 </td>
   <td style="text-align:left;"> 2.31e-04 </td>
   <td style="text-align:right;"> 9 </td>
   <td style="text-align:right;"> 5 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> DGCR6 </td>
   <td style="text-align:left;"> 1.24e-03 </td>
   <td style="text-align:right;"> 48 </td>
   <td style="text-align:right;"> 120 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> PRODH </td>
   <td style="text-align:left;"> 1.24e-03 </td>
   <td style="text-align:right;"> 48 </td>
   <td style="text-align:right;"> 120 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> CSF2RA </td>
   <td style="text-align:left;"> 1.99e-03 </td>
   <td style="text-align:right;"> 8 </td>
   <td style="text-align:right;"> 7 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div><div class="column" style="width:30%;">
<p><img data-src="plots/venn.png" class="absolute" style="top: 250px; right: 0px; width: 350px; "></p>
</div>
</div>
</div>
</section></section>
<section>
<section id="what-next" class="title-slide slide level1 center">
<h1>What next</h1>
<p>More analysis (e.g.&nbsp;Pathway analysis) and more context (e.g.&nbsp;what tissue is gene expressed?)</p>
<ul>
<li class="fragment">Databases (GTEx, Human Protein Atlas, TCGA/cbioportal, CCLE)</li>
</ul>
</section>
<section id="pathway-analysis" class="slide level2 mid">
<h2>Pathway analysis</h2>
<ul>
<li class="fragment">Size reduction: 100’s-1000’s of genes ➙ dozens of pathways</li>
<li class="fragment">Tells a biological story</li>
</ul>
<div class="fragment">
<p><span style="font-weight:1000">Many</span> different methods</p>
<div class="fragment fade-out">
<p><img data-src="images/ORA.png" class="absolute" style="left: 30px; bottom: 70px; "></p>
</div>
<div class="fragment" style="line-height:0.8">
<p><span style="font-weight:1000">Many</span> annotation (gene groupings)</p>
<ul>
<li class="fragment">KEGG</li>
<li class="fragment">Reactome</li>
<li class="fragment">Wikipaths</li>
<li class="fragment">Gene ontology (GO)</li>
</ul>
</div>
</div>
</section>
<section id="performing-pathway-analysis" class="slide level2 smaller">
<h2>Performing pathway analysis</h2>
<p>Webtools:</p>
<p><a href="https://maayanlab.cloud/Enrichr/">Enrichr</a>, <a href="http://www.pantherdb.org/">Panther</a>, <a href="https://david.ncifcrf.gov/">DAVID</a>, …</p>
<div class="fragment">
<p>R:</p>
<div class="cell columns column-output-location" data-hash="Intro_to_bioinformatics2023_cache/revealjs/pathway_analysis_table_07d443121cf7205de4286cea2051ae0d">
<div class="column">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb3-3"><a href="#cb3-3"></a></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="fu">load</span>(<span class="st">"data/wholegene_dup0.05.Rdata"</span>)</span>
<span id="cb3-5"><a href="#cb3-5"></a>entrezid<span class="ot">&lt;-</span><span class="fu">mapIds</span>(org.Hs.eg.db, </span>
<span id="cb3-6"><a href="#cb3-6"></a>                 <span class="at">keys=</span> dup.sig<span class="sc">$</span>Gene_symbol, </span>
<span id="cb3-7"><a href="#cb3-7"></a>                 <span class="at">column=</span><span class="st">"ENTREZID"</span>, </span>
<span id="cb3-8"><a href="#cb3-8"></a>                 <span class="at">keytype=</span><span class="st">"SYMBOL"</span>)</span>
<span id="cb3-9"><a href="#cb3-9"></a>entrezid<span class="ot">&lt;-</span>entrezid[<span class="sc">!</span><span class="fu">is.na</span>(entrezid)]</span>
<span id="cb3-10"><a href="#cb3-10"></a></span>
<span id="cb3-11"><a href="#cb3-11"></a></span>
<span id="cb3-12"><a href="#cb3-12"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> entrezid,</span>
<span id="cb3-13"><a href="#cb3-13"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb3-14"><a href="#cb3-14"></a>                <span class="at">ont           =</span> <span class="st">"BP"</span>,</span>
<span id="cb3-15"><a href="#cb3-15"></a>                <span class="at">pvalueCutoff  =</span> <span class="dv">1</span>,</span>
<span id="cb3-16"><a href="#cb3-16"></a>                <span class="at">qvalueCutoff  =</span> <span class="dv">1</span>,</span>
<span id="cb3-17"><a href="#cb3-17"></a>                <span class="at">readable      =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-18"><a href="#cb3-18"></a></span>
<span id="cb3-19"><a href="#cb3-19"></a></span>
<span id="cb3-20"><a href="#cb3-20"></a><span class="fu">barplot</span>(ego, <span class="at">color=</span><span class="st">"pvalue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div><div class="column fragment">
<div class="cell-output-display">
<p><img data-src="Intro_to_bioinformatics2023_files/figure-revealjs/pathway_analysis_table-1.png" width="960"></p>
</div>
</div>
</div>
</div>
</section>
<section id="pathway-analysis---deletions" class="slide level2 smaller">
<h2>Pathway analysis - Deletions</h2>
<p>More interesting enrichment analysis for significant deletions:</p>

<img data-src="Intro_to_bioinformatics2023_files/figure-revealjs/dels-1.png" width="960" class="r-stretch"></section>
<section id="utlising-public-data" class="slide level2">
<h2>Utlising public data</h2>
<p><span style="color:#068de9"><strong>Human Protein Atlas, GTEx</strong></span>, TCGA and CCLE</p>

<img data-src="Intro_to_bioinformatics2023_files/figure-revealjs/hpar-1.png" width="960" class="r-stretch"></section>
<section id="utlising-public-data-1" class="slide level2">
<h2>Utlising public data</h2>
<p>Human Protein Atlas, GTEx, <span style="color:#068de9"><strong>TCGA</strong></span> and CCLE</p>

<img data-src="Intro_to_bioinformatics2023_files/figure-revealjs/TCGA-1.png" width="960" class="r-stretch"></section>
<section id="utlising-public-data-2" class="slide level2">
<h2>Utlising public data</h2>
<p>Human Protein Atlas, GTEx, TCGA and <span style="color:#068de9"><strong>CCLE</strong></span></p>

<img data-src="Intro_to_bioinformatics2023_files/figure-revealjs/cell_line-1.png" width="960" class="r-stretch"></section>
<section id="utlising-public-data-3" class="slide level2">
<h2>Utlising public data</h2>
<p>Human Protein Atlas, GTEx, TCGA and <span style="color:#068de9"><strong>CCLE</strong></span></p>

<img data-src="Intro_to_bioinformatics2023_files/figure-revealjs/CCLE_2-1.png" width="912" class="r-stretch"></section>
<section id="ranking-potential-candidates" class="slide level2">
<h2>Ranking potential candidates</h2>

<img data-src="Intro_to_bioinformatics2023_files/figure-revealjs/unnamed-chunk-4-1.png" width="384" class="r-stretch"></section>
<section id="section-1" class="slide level2 final">
<h2></h2>
<p><img data-src="images/final.png" class="absolute" style="top:-10px; right:-200px; scale:1.25"></p>
<!-- [Questions]{style="font-size:2em;top:20px"} -->
<div style="color:#0666a8; position: absolute;font-size: 160px;top: 160px; text-transform: uppercase;left: -80px;">
<p>Questions?</p>
</div>
<div style="position:absolute; bottom:40px; font-size: 0.6em">
<p><span style="color: #068de9;font-size:1.5em"><em>Lecture slides and code:</em></span> https://github.com/wigge206/BBiomedSc2023</p>
<p><span style="color: #068de9;font-size:1.5em"><em>Email:</em></span> george.wiggins@otago.ac.nz</p>
</div>
<div class="footer footer-default">

</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="Intro_to_bioinformatics2023_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto-reveal',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>